bin/node/runtime/src/lib.rs

```diff
@@ -642,9 +642,19 @@ impl Get<Option<(usize, ExtendedBalance)>> for OffchainRandomBalancing {
 	}
 }
 
-impl onchain::Config for Runtime {
-	type Accuracy = Perbill;
-	type DataProvider = <Self as pallet_election_provider_multi_phase::Config>::DataProvider;
+pub struct OnChainSequentialPhragmen;
+impl onchain::ExecutionConfig for OnChainSequentialPhragmen {
+	type System = Runtime;
+	type Solver = SequentialPhragmen<
+		AccountId,
+		pallet_election_provider_multi_phase::SolutionAccuracyOf<Runtime>,
+	>;
+	type DataProvider = <Runtime as pallet_election_provider_multi_phase::Config>::DataProvider;
+}
+
+impl onchain::BoundedExecutionConfig for OnChainSequentialPhragmen {
+	type VotersBound = ConstU32<20_000>;
+	type TargetsBound = ConstU32<2_000>;
 }
```

Comment:
```suggestion
pub struct OnchainSeqPhragmen;
impl onchain::Config for OnchainSeqPhragmen {
	type AccountId = AccountId;
	type BlockNumber = BlockNumber; 
	type Solver = SequentialPhragmen<
		Self::AccountId,
		SolutionAccuracyOf<Runtime>,
	>;
	type DataProvider = <Runtime as pallet_election_provider_multi_phase::Config>::DataProvider;
}

impl onchain::BoundedConfig for OnchainSeqPhragmen {
	type VotersBound = ConstU32<20_000>;
	type TargetsBound = ConstU32<2_000>;
}
```

Not a fan of having `type System` there anymore. Conceptually, a `onchain::Config` does not need to assume the existence of a system pallet, and should be able to work individually. Having `type System` is just as bad as my initial bad decision of having `trait onchain::Config: frame_system::Config` :)

Also recommended shorter names.

Reply:
Update: I see that we also have the issue of needing to update weight in the function. This in itself needs to be fixed, but at this point I don't want to stretch this PR too much. I would revert to `onchain::Config: system::Config` for now, or leave it as-is, and I will formulate a new issue about how to benchmark and weigh these onchain executions.

Reply:
Is it not needed for weight calculation and submission though?

Reply:
I would think it's better to keep it as is in this case, because then once the new solution for the weight update is implementation, we would just remove the `type System`. It also means that a pallet could use different `NposSolver` for different fields if that is needed.

Reply:
Sure, I agree.